packages/block-library/src/navigation-link/edit.js

```diff
@@ -242,6 +243,37 @@ export default function NavigationLinkEdit( {
 	const itemLabelPlaceholder = __( 'Add link…' );
 	const ref = useRef();
 
+	const isPostType =
+		( kind && 'post-type' === kind ) || ( type && 'post' === type );
+
+	const hasId = id && ! isNaN( id );
```

Comment:
Why the `isNaN` check is needed?

Reply:
I am not sure of that myself — I added it in an attempt to keep this check consistent with what's done in index.php to decide whether or not a link should be rendered:

https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/navigation-link/index.php#L106-L109